url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/server/genericServer/LocalStringServerSocket.java

```diff
@@ -0,0 +1,422 @@
+     *
+     * @throws ConnectException If stringport name is not found, or is EOF,
+     *                          or if its connect/accept queue is full.
+     * @throws IllegalArgumentException If name is null
+     */
+    public static LocalStringConnection connectTo(String name)
+        throws ConnectException, IllegalArgumentException
+    {
+        return connectTo (name, new LocalStringConnection());
+    }
+
+    /**
+     * Find and connect to stringport with this name.
+     * Intended to be called by client thread.
+     * Will block-wait until the server calls accept().
+     *
+     * @param name Stringport server name to connect to
+     * @param client Existing unused connection object to connect with
+     *
+     * @throws ConnectException If stringport name is not found, or is EOF,
+     *                          or if its connect/accept queue is full.
+     * @throws IllegalArgumentException If name is null, client is null,
+     *                          or client is already peered/connected.
+     *
+     * @return client parameter object, connected to a LocalStringServer
+     */
+    public static LocalStringConnection connectTo(String name, LocalStringConnection client)
+        throws ConnectException, IllegalArgumentException
+    {
+        if (name == null)
+            throw new IllegalArgumentException("name null");
+        if (client == null)
+            throw new IllegalArgumentException("client null");
+        if (client.getPeer() != null)
+            throw new IllegalArgumentException("client already peered");
+
+        if (! allSockets.containsKey(name))
+            throw new ConnectException("LocalStringServerSocket name not found: " + name);
+
+        LocalStringServerSocket ss = (LocalStringServerSocket) allSockets.get(name);       
+        if (ss.isOutEOF())
+            throw new ConnectException("LocalStringServerSocket name is EOF: " + name);
+
+        LocalStringConnection servSidePeer;
+        try
+        {
+            servSidePeer = ss.queueAcceptClient(client);
+        }
+        catch (ConnectException ce)
+        {
+            throw ce;
+        }
+        catch (Throwable t)
+        {
+            ConnectException ce = new ConnectException("Error queueing to accept for " + name);
+            ce.initCause(t);
+            throw ce;
+        }
+
+        // Since we called queueAcceptClient, that server-side thread may have woken
+        // and accepted the connection from this client-side thread already.
+        // So, check if we're accepted, before waiting to be accepted.
+        //
+        synchronized (servSidePeer)
+        {
+            // Sync vs. critical section in accept
+
+            if (! servSidePeer.isAccepted())
+            {
+                try
+                {
+                    servSidePeer.wait();  // Notified by accept method
+                }
+                catch (InterruptedException e) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)